dds/idl/marshal_generator.cpp

```diff
@@ -909,6 +909,366 @@ namespace {
     }
   }
 
+#if OPENDDS_HAS_MAP
+  void gen_map_i(
+    UTL_ScopedName* tdname, AST_Map* map, bool nested_key_only, const FieldInfo* anonymous = 0)
+  {
+    // be_global->add_include("dds/DCPS/Util.h");
+    // be_global->add_include("dds/DCPS/Serializer.h");
+    if (anonymous) {
+      map = dynamic_cast<AST_Map*>(anonymous->type_);
+    }
+    const std::string named_as = anonymous ? anonymous->scoped_type_ : scoped(tdname);
+    RefWrapper base_wrapper(map, named_as, "map");
+    base_wrapper.nested_key_only_ = nested_key_only;
```

Comment:
`RefWrapper` needs to be finalized for it can properly decide what it needs to return. Otherwise it can cause an assert like it's doing now in a least on of the CI jobs.

```suggestion
    RefWrapper base_wrapper(map, named_as, "map");
    base_wrapper.nested_key_only_ = nested_key_only;
    base_wrapper.done();
```